### PR TITLE
Voltdb helm chart

### PR DESCRIPTION
**Pull request by akshaypatidar1999**
### Summary

Voltdb helm chart

### CODE REVIEW

File: charts/voltdb/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+# Voltdb Helm Chart
+
+VoltDB is a commercial in-memory DBMS that is scalable and ACID-compliant.
+
+[Overview of Voltdb](https://www.voltactivedata.com/)
+
+## Introduction
+
+This chart bootstraps an [Voltdb](https://hub.docker.com/r/voltdb/voltdb-community) deployment on a [Kubernetes](https://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
```
Comment (yash-saxena-d11) [inline charts/voltdb/README.md:11]: ```a``` instead of ```an``` would be better.

File: charts/voltdb/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+| `extraDeploy`       | Array of extra objects to deploy with the release                     | `[]`            |
+
+### Voltdb parameters
+
+| Name                                                      | Description                                                                     | Value                     |
+| --------------------------------------------------------- | ------------------------------------------------------------------------------- | ------------------------- |
+| `image.registry`                                          | Voltdb image registry                                                           | `docker.io`               |
+| `image.repository`                                        | Voltdb image repository                                                         | `voltdb/voltdb-community` |
+| `image.tag`                                               | Voltdb image tag (immutable tags are recommended)                               | `9.2.1`                   |
+| `image.pullPolicy`                                        | image pull policy                                                               | `IfNotPresent`            |
```
Comment (yash-saxena-d11) [inline charts/voltdb/README.md:70]: We can keep capitalise the I in image

File: charts/voltdb/README.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+| `extraEnvVars`                                            | Extra environment variables to be set on voltdb container                       | `[]`                      |
+
+### Statefulset parameters
+
+| Name                                 | Description                                   | Value           |
+| ------------------------------------ | --------------------------------------------- | --------------- |
+| `replicaCount`                       | Number of Voltdb replicas                     | `1`             |
+| `affinity`                           | Affinity for pod assignment                   | `{}`            |
+| `nodeSelector`                       | Node labels for pod assignment                | `{}`            |
+| `updateStrategy.type`                | updateStrategy for Voltdb statefulset         | `RollingUpdate` |
```
Comment (yash-saxena-d11) [inline charts/voltdb/README.md:108]: Uppercase U would be better

File: charts/voltdb/templates/headless-svc.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+apiVersion: v1
```
Comment (yash-saxena-d11) [inline charts/voltdb/templates/headless-svc.yaml:1]: should a headless service always be created?

Comment (akshaypatidar1999) [reply charts/voltdb/templates/headless-svc.yaml:1]: Yes, this is needed for voltdb cluster formation

**Review by yash-saxena-d11 — DISMISSED** [review]

File: charts/voltdb/values.yaml [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+  ## @param voltdb.systemSettings.snapshot.priority Snapshot priority
+  ## @param voltdb.systemSettings.elastic.duration Time each rebalance transaction will take, in milliseconds
+  ## @param voltdb.systemSettings.elastic.throughput Number of MBs per second that will be processed by the rebalance transactions
+  ## @param voltdb.systemSettings.query.timeout Query timeout in milliseconds
+  ## @param voltdb.systemSettings.procedure.logInfo Log info whenever a task runs for more than specified time in milliseconds
+  ## @param voltdb.systemSettings.resourceMonitor.frequency Frequency at which resource limits are checked
+  ## @param voltdb.systemSettings.resourceMonitor.memoryLimit.size Maximum allowable resident set size
+  ## @param voltdb.systemSettings.resourceMonitor.memoryLimit.alert Alert thresold
+  ##
+  systemSettings:
```
Comment (yogesh-badke) [inline charts/voltdb/values.yaml:149]: Are there other settings that are not exposed by chart but are available? if yes, we need to have a provision to pass them as is like we are doing in other charts

Comment (akshaypatidar1999) [reply charts/voltdb/values.yaml]: There are other available [configurations](https://docs.voltdb.com/UsingVoltDB/ConfigStructure.php). 
`deployment.xml` is not generated using `voltdb.deployment` in `values`. Some default configurations are set in `values.yaml` and additional configurations can be provided.

**Review by yogesh-badke — APPROVED** [review]

**Review by yogesh-badke — APPROVED** [review]

**Pull request merged**